File: Readability/nehlinshanila-cs50-problems-2020-x-readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");



    int letter = 0;
    int words = 0;
    int sentence = 0;
    int len = strlen(s);


    for ( int i = 0; i < len; i++)

  {
   if (isalpha(s[i]))

   {
       letter++;
   }

   if ((i == 0 && s[i] != ' ') || (i != len - 1 && s[i] == ' ' && s[i + 1] != ' '))
   {
       words++;
   }

   if (s[i] == '!' || s[i] == '?' || s[i] == '.')
   {
       sentence++;
   }
  }

     float L = (100 * (letter / (float) words));
     float S = (100 * (sentence / (float) words));
     int index = round(0.0588 * L - 0.296 * S - 15.8);
     
    if (index < 1)
    {
        printf("Before grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16 +\n");
    }
    else 
    {
        printf("Grade %i\n", index);
    }
 
}

File: sentimental-readability/readability.py
from cs50 import get_string

s = get_string("Text: ")

letters = 0
sentence = 0
words = 0


for char in s:
   if char.isalpha():
      letters += 1
   if char.isspace():
      words += 1
   if char in ['.', '!', '?']:
      sentence += 1
      
words += 1
L = (letters / words) * 100
S = (sentence / words) * 100
index = int((0.0588 * L - 0.296 * S - 15.8) + 0.5)

if index < 1:
   print ("Before Grade 1")
elif index >= 16:
   print ("Grade 16+")
else:
   print (f"Grade, {index}")
